content/ADR-283-explorer-client-scenes-signed-fetch.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: adr
+adr: 283
+title: Explorer Client Scenes Signed Fetch
+date: 2025-11-07
+status: Draft
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - lpetaccio
+  - pentreathm
+  - nachomazzara
+  - gonpombo8
+---
+
+## Abstract
+
+This document describes how the Signed Fetch functionality works in the Explorer Client for Decentraland scenes. It details the structure and purpose of the scene-specific metadata headers that are included when scenes perform HTTP requests. The Explorer Client extends the Signed Fetch mechanism defined in [ADR-44](/adr/ADR-44) with additional contextual metadata including scene identity, parcel location, and realm information, enabling external services to verify the origin and context of requests.
+
+## Context, Reach & Prioritization
+
+### Context
+
+The Explorer Client runs scenes that interact with external HTTP services. When scenes make authenticated requests on behalf of users, additional context beyond the basic Signed Fetch mechanism is valuable for external services. The Explorer Client includes scene-specific metadata in signed requests to provide:
+
+- Scene identification and location (parcel coordinates)
+- Realm information (where the user is connected)
+- Environment context (production, staging, development)
+- Request body integrity verification
+
+This additional metadata enables external services to verify not only the user's identity but also the scene context from which the request originates.
+
+### Reach
+
+This specification impacts:
+
+- All scenes running in the Explorer Client that make HTTP requests
+- Services that need to verify requests originated from specific scenes, locations or realms
+
+## Specification
+
+### Overview
+
+The Explorer Client Signed Fetch functionality extends [ADR-44](/adr/ADR-44) by including scene-specific metadata in the signature. This metadata is included in the `X-Identity-Metadata` header and is part of the signed payload, ensuring its integrity and authenticity.
```

Comment:
Should we add something here to mention that the metadata can not be edited by the creators using the library and it is automatically set by the client?

Reply:
Done, would you mind checking the changes?